src/services/Sso.php

```diff
@@ -0,0 +1,186 @@
+                          'Content-Type: application/json',
+                          'Accept: application/json');
+            curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);    
+        }
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+        if ($method == "POST") {
+            curl_setopt($curl, CURLOPT_POST, true);
+            if($postType == 'json'){
+              curl_setopt($curl,CURLOPT_POSTFIELDS,json_encode($requestData));
+            }else{
+              curl_setopt($curl,CURLOPT_POSTFIELDS,http_build_query($requestData));
+            }
+        } elseif ($method == "GET") {
+            curl_setopt($curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+        }
+        $response = curl_exec($curl);
+        curl_close($curl);
+        return json_decode($response);
+    }
+   function afterDeleteUser($userName){        
+        $forumApiKey  = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumApiKey',false);
+        $forumUrl     = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl',false);
+        $postData     = array(
+                          'type'      =>'json',
+                          'apikey'    => $forumApiKey,
+                          'massaction'=> 'decline_mem',
+                          'usernames' => $userName);
+        $RequestUrl   =  $forumUrl."/register";
+        $result       = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json');
+    }
+    function afterLogOut(){  
+      if(isset($_COOKIE['forumLogoutToken'])){
+        $forumUrl     = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl',false);
+        echo '<img src='.$forumUrl.'"/register/logout?authtoken='.$_COOKIE['forumLogoutToken'].'" border="0" width="0" height="0" alt="" id="imageTag">';  
+        Websitetoolboxforum::getInstance()->sso->resetCookieOnLogout();  
+      }
+    }
+    function resetCookieOnLogout(){
+      setcookie('forumLogoutToken', 0, time() - 3600, "/");
+      setcookie('forumLoginUserid', '', time() - 3600, "/");
+      setcookie('loginRemember', '', time() - 3600, "/");
+   }
+   function renderJsScriptEmbedded($forumUrl){ 
+        $js = <<<JS
+      (  
+       function renderEmbeddedHtmlWithAuthtoken()
+      {  var embedUrl  = "{$forumUrl}";    
+        var wtbWrap = document.createElement('div');
+        wtbWrap.id = "wtEmbedCode";
+        var embedScript = document.createElement('script');
+        embedScript.id = "embedded_forum";
+        embedScript.type = 'text/javascript';
+        var wtbToken = getCookie();
+        if(typeof wtbToken != 'undefined' && wtbToken){
+          embedUrl += "/js/mb/embed.js?authtoken="+wtbToken;
+        } 
+        embedScript.src = embedUrl; 
+        wtbWrap.appendChild(embedScript); 
+        document.getElementById('embedForum').appendChild(embedScript);
+
+      })();
+JS;
+      return $js ;
+    }   
```

Comment:
Please set the indentation of above function.

Reply:
done